### ISSUE TITLE
Error in controller tests after minitest 5.7.0 update

### ISSUE THREAD

**Issue by partydrone** [blowmage/minitest-rails#181]
Given the following test in my controller:

``` ruby
describe ManufacturersController do
  it "deletes a manufacturer" do
    manufacturer = Factory(:manufacturer)
    -> {
      delete :destroy, id: manufacturer.id
    }.must_change "Manufacturer.count", -1
    must_redirect_to root_path
  end
end
```

This works just fine with minitest 5.6.1. After updating to minitest 5.7.0, I get the following error:

```
ManufacturersController#test_0006_deletes a manufacturer:
NoMethodError: private method `binding' called for nil:NilClass
    test/controllers/manufacturers_controller_test.rb:76:in `block (2 levels) in <top (required)>'
```

This also happens when testing my create action. I think the `lambda {}.must_change` syntax is what's causing the problem, but `assert_difference` still works.

Is there a better way to write my tests, or is this a bug?

**Comment by metaskills** [comment]
It might be a bug here. @blowmage I just ran all the mintiest-spec-rails tests using the latest 5.7.0 and everything looks good on my end. Will watch this to see if something has to change.

**Comment by blowmage** [comment]
@partydrone Thank you for submitting this and making me aware of the problem. The project's test coverage does fail on the latest minitest. I'll release a new version that fixes this very soon.

**Comment by blowmage** [comment]
@metaskills I don't see any `infect_an_assertion` calls in minitest-spec-rails. Are you sure you have test coverage for this?

**Comment by blowmage** [comment]
"Very soon" as defined as "whenever travis decides to build". :)

**Comment by blowmage** [comment]
I have identified a fix, committed it, and am currently waiting on travis-ci to build so I can release. Travis is being seriously slow today though.

**Comment by blowmage** [comment]
Please update to minitest-rails 2.2.0 to fix this. The latest Minitest changed some behavior around passing lambdas/procs to expectations. Because of this the `must_change` expectation needed to declare that is is expected to be used with a block.

Apologies for the error and thank you again for submitting the issue.

**Comment by metaskills** [comment]
@blowmage Thanks... where does `must_change` come from? Is this something you add to Rails?

**Comment by blowmage** [comment]
@metaskills Yes, we add expectations (or aliases) for all rails assertions. Code in question is [here](https://github.com/blowmage/minitest-rails/blob/90f3cfda7834c49841ee3c95b96eeb9394f4a5f1/lib/minitest/rails/expectations.rb#L21).

**Comment by blowmage** [comment]
@partydrone ^^That message a couple back was for you. :)

**Comment by metaskills** [comment]
Word! Thanks Mike! 
![mono-lith](https://cloud.githubusercontent.com/assets/2381/7890913/1028ab90-0617-11e5-890c-acc889dab6ad.png)

**Comment by partydrone** [comment]
Awesome! Thanks for such a quick response.

**Comment by JunichiIto** [comment]
I had the same issue with minitest-rails 2.1.1 and resolved after install 2.2.0. Thanks!
